src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperDescriptorConventions.cs

```
@@ -0,0 +1,81 @@
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/> to check.</param>
+        /// <returns>Whether a <see cref="TagHelperDescriptor"/> represents a view component.</returns>
+        public static bool IsViewComponentDescriptor(TagHelperDescriptor descriptor)
+        {
+            return (descriptor != null &&
+                descriptor.PropertyBag.ContainsKey(ViewComponentNameKey)
+                && descriptor.PropertyBag.ContainsKey(ViewComponentTagHelperNameKey));
+        }
+
+        /// <summary>
+        /// Retrieves the view component name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The short name of the view component represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor))
+            {
+                return null;
+            }
+
+            var viewComponentName = descriptor.PropertyBag[ViewComponentNameKey];
+            return viewComponentName;
+        }
+
+        /// <summary>
+        /// Retrieves the view component tag helper name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The custom type name for a view component represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentTagHelperName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor))
+            {
+                return null;
```

Throw instead of returning `null`